controller/backup_controller.go

```diff
@@ -280,23 +306,18 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 			return err
 		}
 
-		if backupTarget.Spec.BackupTargetURL != "" &&
-			backupVolume != nil && backupVolume.DeletionTimestamp == nil {
-			backupTargetClient, err := newBackupTargetClientFromDefaultEngineImage(bc.ds, backupTarget)
-			if err != nil {
-				log.WithError(err).Warn("Failed to init backup target clients")
-				return nil // Ignore error to prevent enqueue
-			}
+		backupTargetClient, err := newBackupTargetClientFromDefaultEngineImage(bc.ds, backupTarget)
+		if err != nil {
+			log.WithError(err).Warn("Failed to init backup target clients")
+			return nil // Ignore error to prevent enqueue
+		}
+		backupURL := backupstore.EncodeBackupURL(backup.Name, backupVolumeName, backupTargetClient.URL)
```

Comment:
You are right, it can be put inside the condition.
I think I moved it outside for some purpose before, but I might have refactored the code myself after.
It is okay to put them inside the function now.